pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -600,10 +608,16 @@ def __getstate__(self) -> dict:
         items = {k: v.__getstate__() for k, v in self.items() if k != '_extra'}
         return {**d, 'items': items}
 
-    def __setstate__(self, state: dict, map_location: Optional[Union[str, torch.device]] = None) -> None:
+    def __setstate__(
+        self,
+        state: dict,
+        map_location: Optional[Union[str, torch.device]] = None,
+        sync_fn: Optional[Callable] = None
+    ) -> None:
         self.__dict__.update({k: v for k, v in state.items() if k != 'items'})
 
         def setstate(k: str, item: dict) -> Union[ResultMetric, ResultMetricCollection]:
+            nonlocal sync_fn
```

Comment:
This is not doing anything as you aren't modifying the value

```suggestion
```

Reply:
Done !